environs/bootstrap/bootstrap.go

```diff
@@ -452,10 +452,15 @@ func bootstrapIAAS(
 		if err != nil && !errors.IsNotFound(err) {
 			return err
 		}
-		if len(availableTools) != 0 && args.AgentVersion == nil {
-			// If agent version was not specified in the arguments,
-			// we always want the latest/newest available.
-			agentVersion, availableTools = availableTools.Newest()
+		if len(availableTools) != 0 {
+			if args.AgentVersion == nil {
+				// If agent version was not specified in the arguments,
+				// we always want the latest/newest available.
+				agentVersion, availableTools = availableTools.Newest()
+			}
+			for _, tool := range availableTools {
+				ctx.Infof("Located Juju agent version %s for %s at %s", tool.Version, bootstrapArch, tool.URL)
```

Comment:
This is really useful if we want to know which available tools will be selected for the bootstrap and from what URL.